src/test/java/core/basesyntax/strategy/PurchaseStrategyHandlerImplTest.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.strategy;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import org.junit.jupiter.api.Test;
+
+class PurchaseStrategyHandlerImplTest {
+    private FruitTransaction fruitTransaction;
+    private FruitDao fruitDao = new FruitDaoImpl();
+    private StrategyHandler purchaseStrategyHandler = new PurchaseStrategyHandlerImpl(fruitDao);
+
+    @Test
```

Comment:
also add test for case where purchase qty is bigger then amount of fruit in storage